modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.js

```diff
@@ -261,28 +263,60 @@ export function FieldBase({
 		columns: [{fields: [field], size: 12}],
 	}));
 
-	const checkRepetitions = () => {
-		let repetitionsCounter = 0;
+	const checkRepetitions = useMemo(() => {
+		if (repeatable && name) {
+			const currentFieldFieldsets = name.match(FIELDSET_REGEX);
+			const currentFieldsetRepeatNumbers = name.match(
+				FIELDSET_REPEAT_NUMBERS_REGEX
+			);
+
+			if (currentFieldsetRepeatNumbers) {
+				currentFieldsetRepeatNumbers.pop();
+			}
+
+			const visitor = new PagesVisitor(pages);
 
-		const visitor = new PagesVisitor(pages);
+			const repeatableFields = [];
 
-		const newParentInstanceId = parentInstanceId;
+			visitor.visitFields((field) => {
+				const fieldFieldsets = field.name.match(FIELDSET_REGEX);
+				const fieldsetRepeatNumber = field.name.match(
+					FIELDSET_REPEAT_NUMBERS_REGEX
+				);
+
+				if (fieldsetRepeatNumber) {
+					fieldsetRepeatNumber.pop();
+				}
+
+				const isSameFieldset =
+					currentFieldFieldsets &&
+					fieldFieldsets &&
+					currentFieldsetRepeatNumbers &&
```

Comment:
I think in this case it would be better to use `currentFieldsetRepeatIndexes` instead of `currentFieldsetRepeatNumbers`; same suggestion to the other constants wirh `Numbers`

Reply:
Great suggestion @rebsilva, done! Thanks!